shell.nix

```diff
@@ -0,0 +1,8 @@
+{ghc}:
```

Comment:
Do we need shell.nix? Why is it here. I'm asking because I think defaults are OK for now. Or do we need this so that Travis works?